src/frontend/src/optimizer/plan_node/stream_materialize.rs

```diff
@@ -281,7 +282,11 @@ impl StreamMaterialize {
 
     /// Rewrite this plan node into [`StreamSink`] with the given `properties`.
     pub fn rewrite_into_sink(self, properties: WithOptions) -> StreamSink {
-        StreamSink::new(self.input, self.table.to_sink_desc(properties))
+        // TODO(Yuanxin): Deduce sink type here.
+        StreamSink::new(
+            self.input,
+            self.table.to_sink_desc(properties, SinkType::Any),
+        )
```

Comment:
IIUC, it is just an intermediate state of the implementation, but can we remove the `SinkType::Any`? I do not think we need it anywhere

Reply:
will we support a connector that can accept both append-only stream and upsert stream?

Reply:
Oh I have seen the comment, I think we can keep it.

Reply:
> IIUC, it is just an intermediate state of the implementation
Yes. This should be refactored later. But we have to deduce sink type somewhere in our code.